src/production.py

```diff
@@ -103,9 +107,14 @@ def is_mapping_feasible(self, graph: Graph, mapping: Dict[NodeHandle, NodeHandle
             return False
 
         # We still need to check whether the hyperedge in the cell centre has appropriate value,
-        # but it is not in the graph model yet.
-
         # Also we need to verify all the edges are of appropriate type
+        for lhs_edge in self.get_lhs().get_edges():
+            for g_edge in graph.get_edges():
+                if rev_mapping[lhs_edge[0]] == g_edge[0] and rev_mapping[lhs_edge[1]] == g_edge[1]:
+                    if lhs_edge[2]['payload'].kind != g_edge[2]['payload'].kind or \
+                            lhs_edge[2]['payload'].value != g_edge[2]['payload'].value:
+                        return False
+                    break
```

Comment:
I don't mind this way of tackling it, but I believe it does not fulfill production prerequisites. 

![image](https://github.com/kkafar/graph-grammars/assets/50801299/ae85e9d5-27cf-478b-8602-31f59e3f3808)

So we actually need to check whether the `Q` hyperedge has `value` set to `True` (check whether there is edge between `(0, 2)` & `(1, 3)` and they both have appropriate type & value.

And all four edges `(0, 1)`, `(1, 2)`, `(2, 3)`, `(3, 0)` are of type `E`:

```python
for node_a, node_b in it.pairwise(nodes + [nodes[0]):
    if graph.edge_for_handles(node_a.handle, node_b.handle).attrs.kind != 'e':
        return False
```

I would suggest making more extensive use of `reverse_mapping` here (like I did above), keeping it mind that `node[0]` here is the node from `state graph` that maps 1:1 with the `node_0` defined in `lhs` graph, `node[1] <-> node_1` etc.: